api/server/prom_zip_collector.go

```diff
@@ -0,0 +1,80 @@
+package server
+
+import (
+	"github.com/fnproject/fn/api/agent"
+	"github.com/openzipkin/zipkin-go-opentracing"
+	"github.com/openzipkin/zipkin-go-opentracing/thrift/gen-go/zipkincore"
+	"github.com/prometheus/client_golang/prometheus"
+	"strings"
+)
+
+// Each span name is published as a separate Histogram metric
+// Using metric names of the form fn_span_<span-name>_duration_seconds
+var histogramVecMap = make(map[string]*prometheus.HistogramVec)
```

Comment:
is there a reason this is not a field of the `PrometheusCollector` struct?

Also, should we flush this every so often in case nobody is reading (prometheus is dead / not hooked up at all)? otherwise, it seems like this will grow unbounded. the zipkin HTTP collector seems to keep 1000 buffered before overflowing, but it's a 'push' model.

Reply:
Agreed, it's neater for `histogramVecMap` to be a field of the `PrometheusCollector` struct. Done.

This is map of Prometheus metrics, keyed by span name. Since span names are hardcoded, the number of keys is fixed and small. We're continually sending observations to each metric, but the Prometheus client is responsible for flushing old observations, not us.